src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Signing/TrustedSignersCommand.cs

```diff
@@ -0,0 +1,160 @@
+                    Strings.TrustCommandActionDescription,
+                    multipleValues: true);
+
+                CommandOption algorithm = trustedSignersCmd.Option(
+                    "--algorithm",
+                    Strings.TrustCommandAlgorithm,
+                    CommandOptionType.SingleValue);
+
+                CommandOption allowUntrustedRootOption = trustedSignersCmd.Option(
+                    "--allow-untrusted-root",
+                    Strings.TrustCommandAllowUntrustedRoot,
+                    CommandOptionType.NoValue);
+
+                CommandOption owners = trustedSignersCmd.Option(
+                    "--owners",
+                    Strings.TrustCommandOwners,
+                    CommandOptionType.MultipleValue);
+
+                CommandOption verbosity = trustedSignersCmd.Option(
+                    "-v|--verbosity",
+                    Strings.Verbosity_Description,
+                    CommandOptionType.SingleValue);
+
+                CommandOption configFile = trustedSignersCmd.Option(
+                    "--configfile",
+                    Strings.Option_ConfigFile,
+                    CommandOptionType.SingleValue);
+
+                trustedSignersCmd.HelpOption(XPlatUtility.HelpOption);
+                trustedSignersCmd.Description = Strings.TrustCommandDescription;
+
+                trustedSignersCmd.OnExecute(async () =>
+                {
+                    TrustCommand action;
+
+                    if (!command.Values.Any() || string.IsNullOrEmpty(command.Values[0]))
+                    {
+                        action = TrustCommand.List;
+                    }
+                    else if (!Enum.TryParse(command.Values[0], ignoreCase: true, result: out action))
+                    {
+                        throw new CommandLineArgumentCombinationException(string.Format(CultureInfo.CurrentCulture, Strings.Error_UnknownAction, command.Values[0]));
+                    }
+
+                    string name = null;
+
+                    if (command.Values.Count > 1)
+                    {
+                        name = command.Values[1];
+                    }
+
+                    string packagePath = null;
+                    string sourceUrl = null;
+                    string fingerprint = null;
+                    if (command.Values.Count() > 2)
+                    {
+                        if (action == TrustCommand.Author || action == TrustCommand.Repository)
+                        {
+                            packagePath = command.Values[2];
+                        }
+                        else if (action == TrustCommand.Source)
+                        {
+                            sourceUrl = command.Values[2];
+                        }
+                        else if (action == TrustCommand.Certificate)
+                        {
+                            fingerprint = command.Values[2];
+                        }
+                    }
```

Comment:
I think the above logic to retrieve options/arguments for sub-commands can be improved.

Reading from the [docs](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.commandlineutils.commandlineapplication.parent?view=dotnet-plat-ext-1.1), it looks like `CommandLineApplication` class supports sub-commands. I have not played around with the `Microsoft.Extensions.CommandLineUtils` API but here is my visualization of `dotnet nuget trust` command based on my current understanding. 

![IMG_8609](https://user-images.githubusercontent.com/52756182/115905670-442ffd00-a41b-11eb-8276-21ca70a7e8dd.jpg)

If were able to get the sub-commands working, we can improve the user experience by adding different `help` messages for sub-commands. Currently, I see the same `help` message in the spec for all sub-commands. AFAIK, we can add sub-commands as shown below.

``` cs
            app.Command("trust", trustedSignersCmd =>
            {
                trustedSignersCmd.Command("author", authorCommand =>
                {
                    CommandArgument name = authorCommand.Argument("name",
                        Strings.TrustAuthorCommandNameDescription,
                        multipleValues: false);

                    CommandArgument package = authorCommand.Argument("package",
                        Strings.TrustAuthorCommandPackageDescription,
                        multipleValues: true);

                    authorCommand.HelpOption(XPlatUtility.HelpOption);
                    authorCommand.Description = Strings.TrustAuthorCommandDescription;
                });
            }
```

Reply:
I think I have another follow up issue for this one https://github.com/NuGet/Home/issues/10788 where I can address.
Current error handling is already not great, I believe it applies other `dotnet nuget` commands too.

Reply:
Sure, I'll use `registering sub-command` later in https://github.com/NuGet/Home/issues/10788.